Remove learnupon API calls, add filtering for users, export all courses by default

keithgg opened the pull request:
## Description

- Adds date and email filters
- Removes the learnupon API calls and populate data from edX tables
- Exports all courses if none are provided

## Testing instructions

Run the exports as normal. The data should be populated from edx tables so should go a lot faster.

## Testing instructions

 Install the code with the command `pip install git+https://github.com/open-craft/learnupon-exporter.git@keith/openedx-tables`

The base command to run an export is: `python manage.py lms learnupon_export_enrollment_data /edx/src/export "course-v1:edX+DemoX+Demo_Course"`

- To export users based on the email domain you can add the option `--email-domain=example.com` and all the records will contain only users with email addresses that belong to that domain.
- To filter on start date, add the option `--start-date=2020-01-01` (`YYYY-MM-DD` format) and records will be filtered for users that started the course after the provided date.

keithgg committed:
chore: remove learnupon api, get data from openedx tables

- added progress indicators
- formatted with black
- version bump to 0.0.2
- fix division by zero

keithgg committed:
feat: filters for email and course enrollment date

- remove started status
- fixes loading settings from lms.yml and studio.yml

keithgg committed:
feat: added course ids to export

keithgg committed:
fix: use passed_timestamp for completion records

pomegranited commented on learnupon_exporter/management/commands/learnupon_export_enrollment_data.py:
```diff
@@ -113,9 +113,9 @@ def export_enrollment_data(self, csv_file):
 
-            if grade:
+            if grade and grade.passed_timestamp:
                 row["Enrollment Score"] = grade.percent_grade
```
The client said they don't care about score -- could you remove this line?

keithgg replied:
Sure. With the other changes it means that I don't need to use the `PersistentCourseGrade` anymore since `first_block_viewed_at` will determine completion.

pomegranited commented on learnupon_exporter/management/commands/learnupon_export_enrollment_data.py:
```diff
@@ -113,9 +113,9 @@ def export_enrollment_data(self, csv_file):
                 row["Enrollment Status"] = "completed"
                 row["Enrollment Started Date"] = first_block_viewed_at
```
The client said that most records don't have a "Start Date", and "start date comes after the finish date".

Instead of using this value, could you use `enrollment.created date` instead?

keithgg replied:
@pomegranited with this change `Enrollment Created Date` == `Enrollment Start Date`. Should I remove the `Enrollment Created Date` column?

pomegranited replied:
Nah keep it -- I think LearnUpon needs it for the uploads. It's ok if it's got the same value in.

pomegranited commented on learnupon_exporter/management/commands/learnupon_export_enrollment_data.py:
```diff
@@ -113,9 +113,9 @@ def export_enrollment_data(self, csv_file):
                 row["Enrollment Status"] = "completed"
```
The client said "Completed Date missing for several records with Completed status". Could you remove this line?

pomegranited reviewed (commented):
Thanks for looking at this @keithgg :)

I left a few comments.

keithgg committed:
chore: review changes

- remove Score column
- removed checking Grade for completion. Only use first_block_viewed_at
- use module completed date instead of start date

pomegranited commented on learnupon_exporter/management/commands/learnupon_export_enrollment_data.py:
```diff
@@ -22,117 +22,88 @@ class Command(ExportCommand):
+            modules = enrollment.user.studentmodule_set.filter(
+                course_id=enrollment.course_id
+            ).order_by("created")
+            first_block_viewed_at = modules.values_list("created", flat=True).first()
```
I guess you could say "enrollment completed at" `modules.values_list("created", flat=True).last()` instead of `first()`? That way the completed date is as late as possible in the process..

keithgg committed:
feat: export all courses

When no courses are specified via the command line, then ALL courses
are exported into multiple CSV files.

pomegranited reviewed (approved):
:+1: Looks great, thank you @keithgg !

- [x] I tested this by running the command on my devstack, and checking the output of the run on the client's servers.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] Includes documentation for how to install and use this management command.
- [x] I made sure any change in configuration variables is reflected in the corresponding
    client's `configuration-secure` repository.
